### PR TITLE
COSI-11 deploy cosi resources in ci

### PR DESCRIPTION
**Pull request by anurag4DSB**
Based on: https://github.com/scality/cosi/pull/9

Details
- Updated Makefile - added GO env for `go build` command
- update-kustomize-files-for-COSI-driver
- deploy-COSI-resources-in-CI

Next steps:
- Add S3 and IAM to COSI CI
- Added E2E test for COSI

### CODE REVIEW

File: .github/scripts/setup_cosi_resources.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+# Step 5: Run COSI driver
+log_and_run echo "Applying COSI driver manifests..."
+log_and_run kubectl apply -k .
+
+# Step 6: Verify COSI driver Pod Status
+log_and_run echo "Verifying COSI driver Pod status..."
+log_and_run kubectl wait --namespace scality-object-storage --for=condition=ready pod --selector=app.kubernetes.io/name=scality-cosi-driver --timeout=20s
+log_and_run kubectl get pods -n scality-object-storage
+
+log_and_run echo "COSI setup completed successfully."
```
Comment (anurag4DSB) [inline .github/scripts/setup_cosi_resources.sh:51]: Better error handling here inone of the next PRs: https://github.com/scality/cosi/pull/12/files#diff-e6be32d24da470375f8cd18f6984e92d51d955beb465dcf8dd05f6939db9ee01R34

File: kustomize/base/rbac.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+apiVersion: rbac.authorization.k8s.io/v1
```
Comment (anurag4DSB) [inline kustomize/base/rbac.yaml:1]: Changed file name to https://github.com/scality/cosi/pull/12/files#diff-fd177c1f152b6ef718481d85599f72ae7def2ac97c456dc0dee75506404f7b8c

File: kustomize/base/namespace.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+apiVersion: v1
```
Comment (anurag4DSB) [inline kustomize/base/namespace.yml:1]: Changed the file name extension from `yml` to `yaml` in one of my next PRs

**Review by dvasilas — APPROVED** [review]

File: Makefile [hunk 4]
```diff
@@ -1,13 +1,17 @@
 APP_NAME = scality-cosi-driver
 BIN_DIR = ./bin
 
+# 'go env' vars aren't always available in make environments, so get defaults for needed ones
+GOARCH ?= $(shell go env GOARCH)
+IMAGE_NAME ?= ghcr.io/scality/cosi:latest
```
[the remark this answers is not in the document]
Comment (anurag4DSB) [reply Makefile:6]: This is just for CI and local deployment.
I didn't want to deal with all the REV logic yet, may be in the future.

**Review by nicolas2bert — APPROVED** [review]

**Pull request merged**